sync/sync_head.go

```diff
@@ -138,7 +144,20 @@ func (s *Syncer[H]) incomingNetworkHead(ctx context.Context, head H) error {
 	s.incomingMu.Lock()
 	defer s.incomingMu.Unlock()
 
-	err := s.verify(ctx, head)
+	// TODO(@Wondertan): We need to ensure Tail is available before verification for subj init case and this is fine here
```

Comment:
can we perform a sanity check here on the incoming head somehow to ensure we do not get exorbitant height that could trigger a cut of the tail too significantly? (meaning one that is shorter than the pruning window)